core/deployment/src/main/java/io/quarkus/deployment/ConfigBuildTimeConfig.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = ConfigItem.PARENT, phase = ConfigPhase.BUILD_TIME)
+public class ConfigBuildTimeConfig {
+    /**
+     * <p>
+     * Set this to <code>true</code> to read configuration from system properties and environment variables only.
+     * </p>
+     */
+    @ConfigItem(name = "config.sources.system-only", defaultValue = "false")
+    public boolean systemOnly;
+}
```

Comment:
This is confusing. It is build time config... and it isn't clear from this description if this constrains config sources at BOTH build time AND runtime or just runtime or ..

Reply:
No, it only applies to runtime configuration.

Reply:
Clarified the javadoc.